src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -129,36 +137,40 @@
            :else [:div {:style {:textAlign "center"}}
                   [comps/Spinner {:text "Loading Method Configuration..."}]]))
    :component-did-mount
-   (fn [{:keys [props state this]}]
+   (fn [{:keys [state this]}]
      (this :-load-validated-method-config)
      (endpoints/call-ajax-orch
       {:endpoint endpoints/list-methods
        :on-done (fn [{:keys [success? get-parsed-response status-text]}]
-                  (if success?
-                    (swap! state assoc :methods (->> (get-parsed-response)
-                                                     (map #(select-keys % [:namespace :name :snapshotId]))
-                                                     (group-by (juxt :namespace :name))
-                                                     (utils/map-values (partial map :snapshotId))))
-                    ;; FIXME: :error-message is unused
-                    (swap! state assoc :error-message status-text)))}))
+                  (let [response (get-parsed-response)]
+                    (if success?
+                      (swap! state assoc :methods-response response :methods (->> response
+                                                                                  (map #(select-keys % [:namespace :name :snapshotId]))
+                                                                                  (group-by (juxt :namespace :name))
+                                                                                  (utils/map-values (partial map :snapshotId))))
+                      ;; FIXME: :error-message is unused
+                      (swap! state assoc :error-message status-text))))}))
```

Comment:
for this ajax call to list-methods, do you ever use any of the methods that do NOT match the namespace/name stored in the config? If you're not, it would be much more efficient to pass that namespace/name as query params to list-methods, and only return the relevant methods, instead of filtering them out via JS.